src/overlays/actors/ovl_Obj_Chan/z_obj_chan.c

```diff
@@ -31,49 +39,404 @@ const ActorInit Obj_Chan_InitVars = {
+}
+
+void ObjChan_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjChan* this = THIS;
 
-extern ColliderCylinderInit D_80BBAB30;
-extern InitChainEntry D_80BBAB5C[];
+    this->actionFunc(this, globalCtx);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
 
-extern UNK_TYPE D_06000AF0;
-extern UNK_TYPE D_06001960;
-extern UNK_TYPE D_06002358;
+void ObjChan_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjChan* this = THIS;
+    Gfx* dl_opa;
+    Gfx* dl_xlu;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/ObjChan_Init.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    Matrix_RotateY(this->rotation, true);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/ObjChan_Destroy.s")
+    dl_opa = Gfx_CallSetupDL(POLY_OPA_DISP, 0x19U);
+    gSPMatrix(dl_opa, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_LOAD);
+    gSPDisplayList(dl_opa + 1, object_obj_chan_DL_000AF0);
+    POLY_OPA_DISP = dl_opa + 2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BB9A1C.s")
+    dl_xlu = func_8012C2B4(POLY_XLU_DISP);
+    gSPMatrix(dl_xlu, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_LOAD);
+    gSPDisplayList(dl_xlu + 1, object_obj_chan_DL_000A10);
+    POLY_XLU_DISP = dl_xlu + 2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BB9B40.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BB9C08.s")
+    Matrix_StatePush();
+    if (this->stateFlags & OBJCHAN_STATE_ON_FIRE) {
+        func_80BBA930(this, globalCtx);
+    }
+    Matrix_StatePop();
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BB9F24.s")
+void ObjChan_DrawPot(Actor* thisx, GlobalContext* globalCtx) {
+    ObjChan* this = THIS;
+    s32 pad;
+    Gfx* dl;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BBA2FC.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    dl = Gfx_CallSetupDL(POLY_OPA_DISP, 0x19U);
+    gSPMatrix(dl, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_LOAD);
+    gSPDisplayList(dl + 1, object_obj_chan_DL_002358);
+    POLY_OPA_DISP = dl + 2;
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BBA314.s")
+    if (this->stateFlags & OBJCHAN_STATE_ON_FIRE) {
+        func_80BBA930(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BBA488.s")
+void func_80BBA930(ObjChan* this, GlobalContext* globalCtx) {
+    u32 sp4C;
+    Gfx* dl;
+    OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/ObjChan_Update.s")
+    sp4C = globalCtx->gameplayFrames;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/ObjChan_Draw.s")
+    Matrix_RotateY(func_800DFCDC(GET_ACTIVE_CAM(globalCtx)) - this->actor.shape.rot.y - this->rotation + 0x8000, 1);
```

Comment:
```suggestion
    Matrix_RotateY(func_800DFCDC(GET_ACTIVE_CAM(globalCtx)) - this->actor.shape.rot.y - this->rotation + 0x8000, MTXMODE_APPLY);
```
Use this enum to specify changes being made to the matrix